lnwire/features.go

```diff
@@ -678,6 +678,50 @@ func EmptyFeatureVector() *FeatureVector {
 	return NewFeatureVector(nil, Features)
 }
 
+// Record implements the RecordProducer interface for FeatureVector. Note that
+// it uses a zero-value type is used to produce the record, as we expect this
+// type value to be overwritten when used in generic TLV record production.
+// This allows a single Record function to serve in the many different contexts
+// in which feature vectors are encoded. This record wraps the encoding/
+// decoding for our raw feature vectors so that we can directly parse fully
+// formed feature vector types.
+func (fv *FeatureVector) Record() tlv.Record {
```

Comment:
I think we can get around needing this by converting the `RawFeatureVector` `Record` method to satisfy the RecordProducer interface and then just using `Features tlv.OptionalRecordT[tlv.TlvType14, lnwire.RawFeatureVector]` in the struct.

Reply:
Tried it locally & seems to work fine:

[rawFeature.patch](https://github.com/lightningnetwork/lnd/files/14647157/rawFeature.patch)

Reply:
Yeah we can! 

@bitromortac's [comment](https://github.com/lightningnetwork/lnd/pull/8159#discussion_r1528308207) made me realize that long term it'll be nice to have the full feature vector so we can use some of the utility methods that we have on `FeatureVector` when we get to the stage where we have to check the features in the blinded path vs the payment features. 

With raw feature vector, we parse it out and then have to call `NewFeatureVector` afterwards to use things like `HasFeature`. But no strong opinions here tbh!

Reply:
oh cool good point! gotcha 👍 

yeah i dont have strong opinions here either so happy for it to be left as is for those future checks 🎉